### PR TITLE
[mpg123] Add patch for wasm triplet, fix checktypesize

### PR DESCRIPTION
**Pull request by playmer**
Seems that something about the wasm env ended up triggering an error when the CMakelists.txt calls `check_type_size` due to CheckTypeSize not being included, Working on submitting a report to the maintainers.

I attempted to update to 1.32.0 that just released a few hours ago, but it seems there's just issues with that release, I think they accidentally dropped a couple folders when they archived the release, as the install fails even on x64-windows for the same reason (can't install the manual due to the file/folder not existing) I'll report that as well after reporting the `check_type_size` fix.
 
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Please do these steps for proper versioning:
1. revert the changes in `versions/`
2. increment `port-version` in `ports/mpg123/vcpkg.json`, and commit this change.
3. run `vcpkg x-add-version mpg123`, and commit the new changes.

File: versions/m-/mpg123.json [hunk 1]
```diff
@@ -1,10 +1,15 @@
 {
   "versions": [
     {
-      "git-tree": "73f380749ae40814a07c7a0952e94db2970a9b85",
+      "git-tree": "c29e53dfebcdd51171128085c3a2e94272469e95",
```
Comment (jimwang118) [inline versions/m-/mpg123.json:4]:
```suggestion
      "git-tree": "73f380749ae40814a07c7a0952e94db2970a9b85",
```

File: versions/m-/mpg123.json [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
       "version": "1.31.3",
       "port-version": 1
     },
+    {
+      "git-tree": "73f380749ae40814a07c7a0952e94db2970a9b85",
+      "version": "1.31.3",
+      "port-version": 2
+    },
```
Comment (jimwang118) [inline versions/m-/mpg123.json:12]:
```suggestion
    {
      "git-tree": "73f380749ae40814a07c7a0952e94db2970a9b85",
      "version": "1.31.3",
      "port-version": 2
    },
```
Please modify the port-version in `vcpkg.json`, and then use` ./vcpkg x-add-version {portname} --overwrite-version` to update the version.

File: ports/mpg123/fix-checktypesize.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
++++ b/ports/cmake/src/CMakeLists.txt
+@@ -5,6 +5,7 @@ include(CheckFunctionExists)
+ include(CheckIncludeFile)
+ include(CheckIncludeFiles)
+ include(CheckSymbolExists)
++include(CheckTypeSize)
+ include(CMakeDependentOption)
+ include(TestBigEndian)
+ 
+
```
Comment (jimwang118) [inline ports/mpg123/fix-checktypesize.patch:13]: Please add a blank line to the end of the file.

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
If your modification is completed, please revert to "ready for review". I will initiate the code merge.

**Comment by BillyONeal** [comment]
>Working on submitting a report to the maintainers.

Please make sure it's submitted to them. Thanks for the fix!

**Pull request merged**